cpp/include/cudf/strings/detail/copying.hpp

```diff
@@ -54,6 +55,33 @@ std::unique_ptr<cudf::column> copy_slice(
   rmm::cuda_stream_view stream        = rmm::cuda_stream_default,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
 
+/**
+ * @brief Returns a new strings column created by shifting the rows by a specified offset.
+ *
+ * @code{.pseudo}
+ * Example:
+ * s = ["a", "b", "c", "d", "e", "f"]
+ * r1 = shift(s, 2, "_")
+ * r1 is now ["_", "_", "a", "b", "c", "d"]
+ * r2 = shift(s, -2, "_")
+ * r2 is now ["a", "b", "c", "d", "_", "_"]
+ * @endcode
+ *
+ * The caller should set the validity mask in the output column.
```

Comment:
I feel it's not a right way to return an incomplete result and expect to complete it somewhere else. But I don't have a better suggestion for this use case.

Reply:
I didn't catch this, but I think @ttnghia is right. What's keeping us from using a string_scalar with a validity bit?

Reply:
This is just an internal detail function called by `cudf::detail::shift`. 
I didn't want to duplicate the code already in the `shift.cu`.
We do this for various other patterns like `detail::gather`, `detail::scatter`, where the calling function handles the null-mask.